File: prgm.py
# translated by AI

import math

# Input values
E = float(input("EMETTEUR (mW)="))
R = float(input("RECEPT (mW)="))
A = float(input("AttenuationFibre (db/km)="))
M = float(input("LongueurFibre (m)="))
F = float(input("Fibre-Fibre (dB)="))
I = float(input("Emetteur-Fibre (dB)="))
G = float(input("Fibre-Recept (dB)="))

# Calculate total power
T = 10 * math.log10(E / R)
print(f"TOTAL={T}")

# Calculate attenuation by fiber link
A *= M / 1000
print(f"attenuation par lien={A}")

# Calculate remaining power
Y = T - (I + A + G)

# Calculate the number of times to take the integer part
O = int(Y / (A + F))
print(f"Nombre entier à prendre={O}")

# Calculate the remaining power after taking the integer part
V = T - (I + A + G) - ((A + F) * O)
print(f"{I}+{A}+{O}*{A+F}+X+{G}={round(T, 3)}")
print(f"Nombre restant={V}")

# Check if there's enough power for additional fiber
if (V - F) > 0:
    P = ((V - F) * M) / A
    print(f"metres de fibre en plus = {P}")
else:
    P = 0
    print("Nombre restant trop petit")

# Calculate the total length of fiber
L = M + O * M + P
print(f"Longueur totale={L}m")

# Calculate the dispersion parameters
L /= 1000
Y = float(input("INDICE Y="))
R = float(input("BANDE INTRA B0="))
K = R / (L ** Y)
print(f"BANDE MODAL={K}")

S = 1 / (2 * K)
print(f"TM={S} µs")
S *= 1000
print(f"{S} ns")

# Calculate dispersion coefficient
Z = float(input("DC="))
J = float(input("GAMMA=")) / 10
Z *= J * L
print(f"TC={Z} ps")
Z /= 1000
print(f"TC={Z} ns")

# Calculate total dispersion
W = math.sqrt(S ** 2 + Z ** 2)
print(f"TTOT={W} ns")

# Calculate bandwidth
N = 1 / (2 * W)
N *= 1000
print(f"BANDE PASSANTE={N} MHz")

File: prgm.vb
Disp "Saisir"
Disp "emetteur"
Disp "en MILLIWATTS"
Input "EMETTEUR=",E

Disp "Saisir"
Disp "recepteur"
Disp "en MILLIWATTS"
Input "RECEPT=",R

10log(E/R)→T
Disp "TOTAL=",T

Disp "Saisir"
Disp "attenuation fibre"
Disp "sur 1000 mètres"
Disp "en dB"
Input "AttenuationFibre=",A

Disp "Saisir"
Disp "longeur fibre"
Disp "en metres"
Input "LongueurFibre=",M

A*M/1000→A
Disp "attenuation par lien"+versChaîne(A)

Disp "Saisir"
Disp "jonction fibre-fibre"
Disp "en dB"
Input "Fibre-Fibre=",F

Disp "Saisir"
Disp "jonction emetteur-fibre"
Disp "en dB"
Input "Emetteur-Fibre=",I

Disp "Saisir"
Disp "jonction fibre-recepteur"
Disp "en dB"
Input "Fibre-Recept=",G

T-(I+A+G)→Y

Y/(A+F)→O
Disp "Nombre entier à prendre=",O
Disp "Donc ",ent(O)

ent(O)→O

T-(I+A+G)-((A+F)*O)→V
Disp versChaîne(I)+"+"+versChaîne(A)+"+"+versChaîne(O)+"*"+versChaîne(A+F)+"+X+"+versChaîne(G)+"="+versChaîne(arrondir(T,3))
Disp "Nombre restant=",V
Pause 
0→P
If (V-F)>0
Then
(((V-F)*M)/A)→P
Disp "metres de fibre en plus =",P
Else
Disp "Nombre restant trop petit"
0→P
End
M+O*M+P→L
Disp "Longueur totale=",L
Pause 
L/1000→L
Input "INDICE Y=",Y
Disp "EN MHZ.KM"
Input "BANDE INTRA B0=",R
R/(L^Y)→K
Disp "BANDE MODAL="
Disp K
Disp "DONC DISPERSION MODAL="
1/(2*K)→S
Disp "TM="
Disp S
Disp "EN US"
S*1000→S
Disp S
Disp "EN NS"
Pause 
Disp "COEF DISPERSION"
Disp "EN PS/(NM*KM)"
Input "DC=",Z
Disp "LARGEUR SPECTRALE EN A"
Disp "10A=1NM"
Input "GAMMA=",J
J/10→J
Z*J*L→I
Disp "TC="
Disp I
Disp "EN PS"
I/1000→I
Disp I
Disp "EN NS"
Pause 
√((S^2)+(I^2))→W
Disp "TTOT="
Disp W
Disp "EN NS"
1/(2*W)→N
N*1000→N
Disp "BANDE PASSANTE"
Disp N
Disp "EN MHZ"
